core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,91 @@
+  _getOpportunityFromDict(opportunityType, opportunityDict) {
+    if (
+      opportunityType === 'voiceover' ||
+      opportunityType === 'translation') {
+      return new ExplorationOpportunitySummary(opportunityDict.id,
+        opportunityDict.topic_name, opportunityDict.story_title,
+        opportunityDict.chapter_title, opportunityDict.content_count,
+        opportunityDict.translation_counts);
+    } else if (opportunityType === 'skill') {
+      return new SkillOpportunity(
+        opportunityDict.id, opportunityDict.skill_description,
+        opportunityDict.topic_name, opportunityDict.question_count);
+    }
+  }
+
+  _fetchOpportunities(opportunityType, params, successCallback) {
+    this.http.get(this.urlInterpolationService.interpolateUrl(
+      this.urlTemplate, { opportunityType }
+    ), {
+      params
+    }).toPromise().then((data: any) => {
+      let opportunities = [];
+      for (let index in data.opportunities) {
+        opportunities.push(this._getOpportunityFromDict(
+          opportunityType, data.opportunities[index]));
       }
-    };
-    var _fetchOpportunities = function(
-        opportunityType, params, successCallback) {
-      return $http.get(
-        UrlInterpolationService.interpolateUrl(
-          urlTemplate, {opportunityType: opportunityType}
-        ), {
-          params: params
-        }).then(function(response) {
-        var data = response.data;
-        var opportunities = [];
-        for (var index in data.opportunities) {
-          opportunities.push(_getOpportunityFromDict(
-            opportunityType, data.opportunities[index]));
-        }
+      if (successCallback) {
         successCallback(opportunities, data.next_cursor, data.more);
-      });
-    };
-    return {
-      fetchSkillOpportunities: function(cursor, successCallback) {
-        var params = {
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_SKILL, params, successCallback);
-      },
-      fetchTranslationOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_TRANSLATION, params, successCallback);
-      },
-      fetchVoiceoverOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_VOICEOVER, params, successCallback);
       }
+    }, (err) => {
+      // console.log(err);
+    });
+  }
+  fetchSkillOpportunities(cursor, successCallback) {
+    let params = {
+      cursor: cursor
     };
-  }]);
+    return this._fetchOpportunities(
+      'skill', params, successCallback);
+  }
+  fetchTranslationOpportunities( languageCode, cursor, successCallback ) {
```

Comment:
Add a new line between two functions. (Just to make a visual separation.)

Reply:
Done